### PR TITLE
Add token generation

### PR DESCRIPTION
**Pull request by CodeReaper**

### CODE REVIEW

File: domains/auth/Auth.API/API/Utilities/TokenIssuer.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+using System.Security.Cryptography;
+using System.Text;
+using API.Options;
+using Microsoft.IdentityModel.Tokens;
+
+namespace API.Utilities;
+
+public class TokenIssuer
+{
+    public static string Issue(TokenOptions options, string userId, DateTime? issueAt = default)
```
Comment (CodeReaper) [inline domains/auth/Auth.API/API/Utilities/TokenIssuer.cs:12]: Add issueAt to interface, and inherit from said interface

File: domains/auth/Auth.API/API/AutoGenerated/Migrations/DataContextModelSnapshot.cs [hunk 2]
```diff
@@ -49,7 +49,7 @@ protected override void BuildModel(ModelBuilder modelBuilder)
 
                     b.HasIndex("ProviderId");
 
-                    b.ToTable("Users");
+                    b.ToTable("Users", (string)null);
```
Comment (robertrosborg) [inline domains/auth/Auth.API/API/AutoGenerated/Migrations/DataContextModelSnapshot.cs:52]: Why?

File: domains/auth/Auth.API/API/Options/OidcOptions.cs [hunk 3]
```diff
@@ -1,6 +1,13 @@
 namespace API.Options;
 
-public record OidcOptions(Uri AuthorityUri, TimeSpan CacheDuration, string ClientId, Uri AuthorityCallbackUri, Uri FrontendRedirectUri)
+#pragma warning disable CS8618
+public class OidcOptions
 {
     public const string Prefix = "Oidc";
+
+    public Uri AuthorityUri { get; init; }
```
Comment (robertrosborg) [inline domains/auth/Auth.API/API/Options/OidcOptions.cs:8]:
```suggestion
    public Uri AuthorityUri { get; init; } = null!;
```
Remove the #pragma

File: domains/auth/Auth.API/API/Options/OidcOptions.cs [hunk 4]
```diff
@@ -1,6 +1,13 @@
...
+    public TimeSpan CacheDuration { get; init; }
+    public string ClientId { get; init; }
```
Comment (robertrosborg) [inline domains/auth/Auth.API/API/Options/OidcOptions.cs:10]:
```suggestion
    public string ClientId { get; init; }  = null!;
```

File: domains/auth/Auth.API/API/Options/OidcOptions.cs [hunk 5]
```diff
@@ -1,6 +1,13 @@
...
+    public Uri AuthorityCallbackUri { get; init; }
```
Comment (robertrosborg) [inline domains/auth/Auth.API/API/Options/OidcOptions.cs:11]:
```suggestion
    public Uri AuthorityCallbackUri { get; init; }  = null!;
```

File: domains/auth/Auth.API/API/Options/OidcOptions.cs [hunk 6]
```diff
@@ -1,6 +1,13 @@
...
+    public Uri FrontendRedirectUri { get; init; }
```
Comment (robertrosborg) [inline domains/auth/Auth.API/API/Options/OidcOptions.cs:12]:
```suggestion
    public Uri FrontendRedirectUri { get; init; }  = null!;
```

File: domains/auth/Auth.API/API/Options/OidcOptions.cs [hunk 7]
```diff
@@ -1,6 +1,13 @@
 namespace API.Options;
 
-public record OidcOptions(Uri AuthorityUri, TimeSpan CacheDuration, string ClientId, Uri AuthorityCallbackUri, Uri FrontendRedirectUri)
+#pragma warning disable CS8618
```
Comment (robertrosborg) [inline domains/auth/Auth.API/API/Options/OidcOptions.cs:3]:
```suggestion
```

File: domains/auth/Auth.API/API/Options/TokenOptions.cs [hunk 8]
```diff
@@ -0,0 +1,13 @@
+namespace API.Options;
+
+#pragma warning disable CS8618
```
Comment (robertrosborg) [inline domains/auth/Auth.API/API/Options/TokenOptions.cs:3]:
```suggestion
```

File: domains/auth/Auth.API/API/Options/TokenOptions.cs [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+public class TokenOptions
+{
+    public const string Prefix = "Token";
+
+    public string Audience { get; init; }
```
Comment (robertrosborg) [inline domains/auth/Auth.API/API/Options/TokenOptions.cs:8]:
```suggestion
    public string Audience { get; init; } = null!;
```

File: domains/auth/Auth.API/API/Options/TokenOptions.cs [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+    public string Issuer { get; init; }
```
Comment (robertrosborg) [inline domains/auth/Auth.API/API/Options/TokenOptions.cs:9]:
```suggestion
    public string Issuer { get; init; }  = null!;
```

File: domains/auth/Auth.API/API/Options/TokenOptions.cs [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+    public TimeSpan Duration { get; init; }
+    public byte[] PrivateKeyPem { get; init; }
```
Comment (robertrosborg) [inline domains/auth/Auth.API/API/Options/TokenOptions.cs:11]:
```suggestion
    public byte[] PrivateKeyPem { get; init; }  = null!;
```

File: domains/auth/Auth.API/API/Options/TokenOptions.cs [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+    public byte[] PublicKeyPem { get; init; }
```
Comment (robertrosborg) [inline domains/auth/Auth.API/API/Options/TokenOptions.cs:12]:
```suggestion
    public byte[] PublicKeyPem { get; init; }  = null!;
```

File: domains/auth/Auth.API/API/Repositories/Data/DataContext.cs [hunk 13]
```diff
@@ -9,14 +9,9 @@ public class DataContext : DbContext, IUserDataContext
 
         public DbSet<User> Users { get; set; }
```
Comment (robertrosborg) [inline domains/auth/Auth.API/API/Repositories/Data/DataContext.cs:10]:
```suggestion
        public DbSet<User> Users { get; set; }  = null!;
```

File: domains/auth/Auth.API/API/Repositories/Data/DataContext.cs [hunk 14]
```diff
@@ -9,14 +9,9 @@ public class DataContext : DbContext, IUserDataContext
...
-        public DataContext(DbContextOptions options, IConfiguration configuration) : base(options)
-        {
-            Configuration = configuration;
-        }
-
-        protected override void OnModelCreating(ModelBuilder modelBuilder)
-        {
-            base.OnModelCreating(modelBuilder);
-        }
+#pragma warning disable CS8618
```
Comment (robertrosborg) [inline domains/auth/Auth.API/API/Repositories/Data/DataContext.cs:12]:
```suggestion
```

**Review by robertrosborg — DISMISSED** [review]

**Review by robertrosborg — APPROVED** [review]

**Pull request merged**